import-export-cli/integration/apiProduct_test.go

```diff
@@ -0,0 +1,774 @@
+
+	args := &apiProductImportExportTestArgs{
+		apiProductProvider:   credentials{username: apiPublisher, password: apiPublisherPassword},
+		ctlUser:              credentials{username: adminUsername, password: adminPassword},
+		apiProduct:           apiProduct,
+		srcAPIM:              dev,
+		destAPIM:             prod,
+		importApisFlag:       false,
+		updateApisFlag:       true,
+		updateApiProductFlag: false,
+	}
+
+	// Import the API Product without importing the dependent APIs to env2 but updating those
+	validateAPIProductExportImportPreserveProvider(t, args)
+}
+
+// Export an API Product with its dependent APIs from one environment and import to another environment freshly as super tenant admin
+// and try to update that API Product (without updating dependent APIs)
+func TestExportImportApiProductAdminSuperTenantUserWithUpdateApiProduct(t *testing.T) {
+	adminUsername := superAdminUser
+	adminPassword := superAdminPassword
+
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	apiPublisher := publisher.UserName
+	apiPublisherPassword := publisher.Password
+
+	dev := apimClients[0]
+	prod := apimClients[1]
+
+	// Add the first dependent API to env1
+	dependentAPI1 := addAPI(t, dev, apiCreator, apiCreatorPassword)
+	publishAPI(dev, apiPublisher, apiPublisherPassword, dependentAPI1.ID)
+
+	// Add the second dependent API to env1
+	dependentAPI2 := addAPIFromOpenAPIDefinition(t, dev, apiCreator, apiCreatorPassword)
+	publishAPI(dev, apiPublisher, apiPublisherPassword, dependentAPI2.ID)
+
+	// Map the real name of the API with the API
+	apisList := map[string]*apim.API{
+		"PizzaShackAPI":   dependentAPI1,
+		"SwaggerPetstore": dependentAPI2,
+	}
+
+	// Add the API Product to env1
+	apiProduct := addAPIProductFromJSON(t, dev, apiPublisher, apiPublisherPassword, apisList)
+
+	args := &apiProductImportExportTestArgs{
+		apiProductProvider:   credentials{username: apiPublisher, password: apiPublisherPassword},
+		ctlUser:              credentials{username: adminUsername, password: adminPassword},
+		apiProduct:           apiProduct,
+		srcAPIM:              dev,
+		destAPIM:             prod,
+		importApisFlag:       true,
+		updateApisFlag:       false,
+		updateApiProductFlag: false,
+	}
+
+	t.Cleanup(func() {
+		apiList := getAPIs(prod, apiCreator, apiCreatorPassword)
+		for _, api := range apiList.List {
+			deleteAPI(t, prod, api.ID, apiCreator, apiCreatorPassword)
+		}
+	})
+
+	// Export the API Product from env 1 and import it to env 2
+	validateAPIProductExportImportPreserveProvider(t, args)
+
+	// Set the --update-api-product flag to update the existing API Product while importing
+	// and make the importApisFlag false
+	args.importApisFlag = false
+	args.updateApiProductFlag = true
+
+	// Re-import the API Product to env 1 while updating it
+	validateAPIProductImportPreserveProviderWithoutCleaningImportedAPIProduct(t, args)
```

Comment:
I understand that you are not doing the cleanup within the function because this is simply an update of the already imported API Product. The previous call to _validateAPIProductExportImportPreserveProvider()_ will take care of the cleanup of the API Product.

However, we shouldnt name this function as  _validateAPIProductImportPreserveProviderWithoutCleaningImportedAPIProduct()_, it leaks to much information regarding its implementation(we dont need to inform people that it does a cleanup or not). We could simply say _validateAPIProductImportUpdate_ or something like that, because it clearly communicates what the function is actually doing. We need the tests to be easy to understand at first glance without needing to look at the function implementations.

Reply:
Sure, will do the changes where applicable.